storm-core/src/clj/backtype/storm/daemon/nimbus.clj

```diff
@@ -346,53 +394,124 @@
       [(.getNodeId slot) (.getPort slot)]
       )))
 
+(defn- get-nimbus-subject []
+  (let [nimbus-subject (Subject.)
+        nimbus-principal (NimbusPrincipal.)
+        principals (.getPrincipals nimbus-subject)]
+    (.add principals nimbus-principal)
+    nimbus-subject))
+
+(defn- get-metadata-version [blob-store key subject]
+  (let [blob-meta (.getBlobMeta blob-store key subject)]
+    (.get_version blob-meta)))
+
+(defn get-key-list-from-blob-store [blob-store]
+  (let [key-iter (.listKeys blob-store (get-nimbus-subject))
+        keys (iterator-seq key-iter)]
+    (if (not-nil? keys)
+      (java.util.ArrayList. keys)
+      [])))
```

Comment:
Here in one place you're returning a java ArrayList, and in the other you're returning a clojure vector.